### PR TITLE
add a new file ratInAMaze

### PR DESCRIPTION
**Pull request by nayanajain**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by nayanajain** [comment]
updated readme file as well...please merge my code.

File: C-Plus-Plus/cp/ratInAMaze.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/*There is a rat who is at the starting point of a matrix and wants to reach
+ to the ending point i.e at the last coordinate of the matrix.
+ 1 denotes the point where it is possible for the mouse to go and
+ 0 denotes that the mouse cannot reach at that point. */
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/ratInAMaze.cpp]: Can you add more details about the Problem here 😅 

Also if possible add a Problem Link on some Online Coding Platform

File: C-Plus-Plus/cp/ratInAMaze.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    }
+
+
+    return false;
+}
+
+
+int main()
+{
+    int m, n;
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/ratInAMaze.cpp]: Use meaningful variable names

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Do these changes as well: 

- Remove extra lines of code 
- Format the Code to make it look readable
- Put Comments above the Code, not beside it

**Comment by nayanajain** [comment]
done the requested changes

File: C-Plus-Plus/README.md [hunk 3]
```diff
@@ -47,6 +47,13 @@
 - [Suduko Solver](cp/SudukoSolver.cpp)
 - [Target Sum triplets](cp/target_sum_triplets.cpp)
 - [Z Algorithm](cp/zalgorithm.cpp)
+- [ArraySub](cp/ARRAYSUB.cpp)
+- [Suduko Solver](cp/SudukoSolver.cpp)
+- [Josephus Problem](cp/Josephus.cpp)
+- [NQueens](cp/NQueens.cpp)
+- [rat in a maze](cp/ratInAMaze.cpp)
+- [Longest Subarray with Zero Sum](cp/ZeroSum.cpp)
+- [Next Largest Permutation](cp/NextPermutation.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:56]: Why so many README changes?

File: C-Plus-Plus/cp/ratInAMaze.cpp [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        cout<<endl;
+    }
+
+
+    }
+
+    return 0;
+}
+
+
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/ratInAMaze.cpp]: Remove extra lines of code

File: C-Plus-Plus/cp/ratInAMaze.cpp [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    {
+        solution[x][y]=1;
+        return true;
+    }
+
+    if(issafetomove(arr,x,y,row,col))
+    {
+        solution[x][y]=1;
+
+//mpving in right direction
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/ratInAMaze.cpp]: Can you remove the typos? Also properly format the Comments to make them look meaningful

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md (hunk 3 above)
Comment (HarshCasper) [reply C-Plus-Plus/README.md:56]: The Question still remains.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Remove unnecessary lines of Code

**Comment by nayanajain** [comment]
I have added only a single line (rat in a maze) in the Readme file ..I am
not able to understand  your change request

**Comment by nayanajain** [comment]
I have added a single line that is name of the file in my readme...I am not able to get why showing so many changes
I have added link of the problem and removed blank lines between the code..please merge my code

If my pr not gets merged, will I get points for issue??please clear my query
Thank you

**Pull request closed without merging**